File: src/2025/day04/src/test/java/ca/terrylockett/aoc2025/day04/TestDay04.java
package ca.terrylockett.aoc2025.day04;

import ca.terrylockett.aoccommon.resources.Resources;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class TestDay04 {

	static final String TEST_FILE_NAME = "test-input.txt";

	static String testInput = "";

	@BeforeAll
	static void setup() {
		testInput = Resources.getInput(TEST_FILE_NAME).orElseThrow();
	}

	@Test
	void testDay04part1() {
		assertEquals(13, Day04Runner.part1(testInput));
	}

	@Test
	void testDay04part2() {
		assertEquals(43, Day04Runner.part2(testInput));
	}

}

File: src/2023/day04/src/test/kotlin/ca/terrylockett/aoc2023/day04/TestDay04.kt
package ca.terrylockett.aoc2023.day04

import ca.terrylockett.aoccommon.resources.Resources
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class TestDay04 {
	companion object {
		const val TEST_FILE_NAME = "test-input.txt"
	}

	@Test
	fun part1() {
		val inputFilePath = Resources.getInputFilePath(TEST_FILE_NAME).orElseThrow()

		assertEquals(13, part1GetPoints(inputFilePath))
	}

	@Test
	fun part2() {
		val inputFilePath = Resources.getInputFilePath(TEST_FILE_NAME).orElseThrow()

		assertEquals(30, part2GetPoints(inputFilePath))
	}
}
